### ISSUE TITLE
Bad function adress in __emit

### ISSUE THREAD

**Issue by samphunter** [pawn-lang/compiler#273]
```C
#include <a_samp>

public OnFilterScriptInit() {
	new Func:a, Func:b;
	
	__emit const.pri TestFunc;
	__emit stor.S.pri a;
	
	#emit CONST.pri  TestFunc
	#emit STOR.S.pri b
	
	printf("%x == %x", _:a, _:b);
        return 1;
}

TestFunc() {
	print("Success!");
	return 1;
}
```

This produces `9D4 == B8`. I assume this is because __emit takes absolute adress while #emit takes relative, but this should be fixed so that both use relative adress.

**Comment by YashasSamaga** [comment]
What is `TestFunc`? I suspect that your test code doesn't even compile.

**Comment by samphunter** [comment]
@YashasSamaga sorry, edited to icnlude TestFunc(). Also it does not seem to be relative adress, as I am trying to account for it and am failing.

**Comment by YashasSamaga** [comment]
Ah, I think that's happening because `TestFunc` is declared later.

**Comment by samphunter** [comment]
@YashasSamaga Yes, putting TestFunc before the emit seems to fix the problem. Though I would say it would be preferable, that the compiler could handle this entirely, just like the #emit version can. Undefined symbol would still be annoying for many usecases, such as my attempt to use function pointers as arguments to other functions.

**Comment by YashasSamaga** [comment]
Actually I was wrong about undefined symbol. That error is applicable to variables only.

The `__emit` syntax seems to be providing the wrong address.

**Comment by Zeex** [comment]
Closed because fixed by #279.
